pkg/webhook/patch.go

```diff
@@ -292,16 +293,78 @@ func addEnvironmentVariable(pod *corev1.Pod, envName, envValue string) *patchOpe
 	return &patchOperation{Op: "add", Path: path, Value: value}
 }
 
-func addSparkConfigMap(pod *corev1.Pod, app *v1beta2.SparkApplication) []patchOperation {
+/*
+	Related issue: https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/issues/216
+	---
+	Previous behavior:
+	  1. Change SPARK_CONF_DIR to /etc/spark/conf (otherwise the default spark.properties will be overwritten)
+	  2. Add VolumeMount for ConfigMap
+	Previous result:
+	  * Effectively mounting files from sparkConfMap to /etc/spark/conf
+	Issue with outcome:
+	  * Spark 2.4.x-3.2.1 does not pick up the spark configuration from /etc/spark/conf or sparkConfigMap
+	---
+	Proposed behavior:
+	  1. Do not change SPARK_CONF_DIR from /opt/spark/conf
+	  2. Change VolumeMount mountPath for `spark.properties` to subPath
+	  3. subPath all other keys from the sparkConfMap spec provided ConfigMap
+	---
+	Edge cases:
+	  * Handling of duplicate `spark.properties` files which carry the configuration for the driver and executor(s)
+ */
+func addSparkConfigMap(pod *corev1.Pod, app *v1beta2.SparkApplication, client kubernetes.Interface) []patchOperation {
 	var patchOps []patchOperation
 	sparkConfigMapName := app.Spec.SparkConfigMap
 	if sparkConfigMapName != nil {
+		// Add Volume populated by the ConfigMap
 		patchOps = append(patchOps, addConfigMapVolume(pod, *sparkConfigMapName, config.SparkConfigMapVolumeName))
-		vmPatchOp := addConfigMapVolumeMount(pod, config.SparkConfigMapVolumeName, config.DefaultSparkConfDir)
-		if vmPatchOp == nil {
-			return nil
+		containerIndex := findContainer(pod)
+		// Note: no error handling is done for `containerIndex == -1` because we anticipate the Pod to handle the Container lifecycle
+		glog.V(2).Infof("SubPath patching existing VolumeMounts: %v", pod.Spec.Containers[containerIndex].VolumeMounts)
```

Comment:
Regarding most logs from this line to the end of the function, I feel it's a bit too verbose for this log level. 

internal logs here should be Debug IMO as long as we have an info level log that provides a bit more context on how the process really affects the end user in terms of what gets mounted and whether the process failed.

For example
```
patching spark.properties from config map NAME to MOUNT_POINT
``` conveys more to the user as an info log

Reply:
Except that we don't patch `spark.properties` but I'm with you. I'll patch the `V(2).Infof` message.